provider_gsuite_test.go

```diff
@@ -0,0 +1,506 @@
+					UserCustomSchemas:      "Custom",
+					serviceAccountKeyJSON:  []byte(`{"type": "service_account"}`),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "invalid config: recurse max depth negative number",
+			args: args{
+				config: GSuiteProviderConfig{
+					ServiceAccountFilePath: "/path/to/google-service-account.json",
+					AdminImpersonateEmail:  "test@example.com",
+					GroupsRecurseMaxDepth:  -1,
+					UserCustomSchemas:      "Custom",
+					serviceAccountKeyJSON:  []byte(`{"type": "service_account"}`),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid config: all options",
+			args: args{
+				config: GSuiteProviderConfig{
+					ServiceAccountFilePath: "/path/to/google-service-account.json",
+					AdminImpersonateEmail:  "test@example.com",
+					GroupsRecurseMaxDepth:  5,
+					UserCustomSchemas:      "Custom",
+					serviceAccountKeyJSON:  []byte(`{"type": "service_account"}`),
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "valid config: no custom schemas",
+			args: args{
+				config: GSuiteProviderConfig{
+					ServiceAccountFilePath: "/path/to/google-service-account.json",
+					AdminImpersonateEmail:  "test@example.com",
+					GroupsRecurseMaxDepth:  5,
+					serviceAccountKeyJSON:  []byte(`{"type": "service_account"}`),
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "valid config: no recurse max depth",
+			args: args{
+				config: GSuiteProviderConfig{
+					ServiceAccountFilePath: "/path/to/google-service-account.json",
+					AdminImpersonateEmail:  "test@example.com",
+					serviceAccountKeyJSON:  []byte(`{"type": "service_account"}`),
+					UserCustomSchemas:      "Custom",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "valid config: fetch groups and user info",
+			args: args{
+				config: GSuiteProviderConfig{
+					ServiceAccountFilePath: "/path/to/google-service-account.json",
+					AdminImpersonateEmail:  "test@example.com",
+					serviceAccountKeyJSON:  []byte(`{"type": "service_account"}`),
+					UserCustomSchemas:      "Custom",
+					FetchGroups:            true,
+					FetchUserInfo:          true,
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := &GSuiteProvider{}
+			if tt.wantErr {
+				assert.Error(t, g.initialize(tt.args.config))
+			} else {
+				assert.NoError(t, g.initialize(tt.args.config))
+			}
+		})
+	}
+}
+
+// getTestCreds gets credentials needed to run tests from environment variables.
+// It will log and skip the test if the required credentials are not set.
+func getTestCreds(t *testing.T) (string, string) {
+	c := os.Getenv(googleCredentialsEnv)
+	if c == "" {
+		t.Logf("skip: must set env var %q to a valid service account key file path", googleCredentialsEnv)
+		t.SkipNow()
+	}
+
+	a := os.Getenv(gsuiteAdminEmailEnv)
+	if a == "" {
+		t.Logf("skip: must set env var %q to a gsuite admin email address", gsuiteAdminEmailEnv)
+		t.SkipNow()
+	}
+
+	return c, a
+}
```

Comment:
I wonder if it'd make sense to move this up to the top of the file, just so it's easier to see what environment variables need to be set.

Reply:
Good idea. I've moved it to the top in ecd50f242cc337cb299a77057107f540291f4c7b.